rpm/kubelet.spec

```diff
@@ -138,11 +138,11 @@ cp -p %SOURCE4 %{_builddir}/
 cp -p %SOURCE6 %{_builddir}/
 %endif
 %setup -c -D -T -a 5 -n cni-plugins
-%setup -c -a 7 -T -n cri-tools
+%setup -c -D -T -a 7 -n cri-tools
```

Comment:
What is the rationale for adding the -D argument here?  The combination of multiple %setup macro calls and the -c, -D, -T and -n arguments makes this a mildly complicated invocation to rationalize without doing some debug runs.  I presume you have done that.

It would be useful for review if you showed an example of what you have debugged the %setup directory structure to be before your change and after your change.

We're also looking to move toward the k/k based build tools which have a dramatically simplified cri-tool specfile: https://github.com/kubernetes/kubernetes/blob/master/build/rpms/cri-tools.spec

Reply:
the "-D" option is not necessary, just to keep the consistent with the previous %setup marco.  details about the %setup marco, can refer to http://ftp.rpm.org/max-rpm/s1-rpm-inside-macros.html

actually, i read below comments from k/k repo, and start to check this repo ；）
https://github.com/kubernetes/kubernetes/blob/master/build/lib/release.sh
```
###############################################################################
# Most of the ::release:: namespace functions have been moved to
# github.com/kubernetes/release.  Have a look in that repo and specifically in
# lib/releaselib.sh for ::release::-related functionality.
###############################################################################
```